.github/workflows/docs-workflow.yaml

```diff
@@ -0,0 +1,72 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: Docs generation for Github Pages
+
+on:
+  push:
+    paths:
+    # Doc updates
+    - 'docs/*'
+    # The action itself
+    - '.github/workflows/docs-workflow.yaml'
+    branches:
+    # Create the docs only when it matters
+    - 'master'
+    - 'release-*'
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+    # Checkout the source
+    - uses: actions/checkout@v2
+      with:
+        path: astarte-device-sdk-csharp
+    # Checkout the docs repository
+    - uses: actions/checkout@v2
+      with:
+        repository: astarte-platform/docs
+        ssh-key: ${{ secrets.DOCS_DEPLOY_KEY }}
+        path: docs
+    - name: Install doxygen
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install doxygen
+    - name: Build Docs
+      working-directory: ./astarte-device-sdk-csharp/docs
+      run: doxygen doxy_config
+    - name: Copy Docs
+      run: |
+        export DOCS_DIRNAME="device-sdks/csharp/$(echo ${{ github.ref }} | sed 's,refs/heads/,,' | sed 's/master/snapshot/g' | sed 's/release-//g')"
```

Comment:
All the other SDKs APIs documentation are saved in the location `device-sdks/SDK_NAME/VERSION/api`.
It's better to keep this convention as it is used for the links of the [SDK documentation](https://docs.astarte-platform.org/device-sdks/common/api_docs.html)